src/tools/r2rdump/R2RReader.cs

```diff
@@ -65,130 +71,177 @@ public unsafe R2RReader(string filename)
                     }
+                    R2RMethods.Add(method);
+                }
+                curParser = allEntriesEnum.GetNext();
+            }
+        }
 
-                    // get the RVAs of the runtime functions for each method
-                    int curOffset = 0;
-                    foreach (R2RMethod method in R2RMethods)
+        /// <summary>
+        /// Get the RVAs of the runtime functions for each method
+        /// </summary>
+        private void ParseRuntimeFunctions(bool[] isEntryPoint, int runtimeFunctionOffset, int runtimeFunctionSize)
+        {
+            int curOffset = 0;
+            foreach (R2RMethod method in R2RMethods)
+            {
+                int runtimeFunctionId = method.EntryPointRuntimeFunctionId;
+                if (runtimeFunctionId == -1)
+                    continue;
+                curOffset = runtimeFunctionOffset + runtimeFunctionId * runtimeFunctionSize;
+                do
+                {
+                    int startRva = NativeReader.ReadInt32(Image, ref curOffset);
+                    int endRva = -1;
+                    if (Machine == Machine.Amd64)
                     {
-                        int runtimeFunctionId = method.EntryPointRuntimeFunctionId;
-                        if (runtimeFunctionId == -1)
-                            continue;
-                        curOffset = runtimeFunctionOffset + runtimeFunctionId * runtimeFunctionSize;
-                        do
-                        {
-                            int startRva = NativeReader.ReadInt32(Image, ref curOffset);
-                            int endRva = -1;
-                            if (Machine == Machine.Amd64)
-                            {
-                                endRva = NativeReader.ReadInt32(Image, ref curOffset);
-                            }
-                            int unwindRva = NativeReader.ReadInt32(Image, ref curOffset);
-
-                            method.RuntimeFunctions.Add(new RuntimeFunction(runtimeFunctionId, startRva, endRva, unwindRva, method));
-                            runtimeFunctionId++;
-                        }
-                        while (runtimeFunctionId < nRuntimeFunctions && !isEntryPoint[runtimeFunctionId]);
+                        endRva = NativeReader.ReadInt32(Image, ref curOffset);
                     }
+                    int unwindRva = NativeReader.ReadInt32(Image, ref curOffset);
+
+                    method.RuntimeFunctions.Add(new RuntimeFunction(runtimeFunctionId, startRva, endRva, unwindRva, method));
+                    runtimeFunctionId++;
                 }
+                while (runtimeFunctionId < isEntryPoint.Length && !isEntryPoint[runtimeFunctionId]);
+            }
+        }
+
+        private void ParseAvailableTypes()
+        {
+            R2RSection instMethodEntryPointSection = R2RHeader.Sections[R2RSection.SectionType.READYTORUN_SECTION_AVAILABLE_TYPES];
```

Comment:
Can you rename this to something more appropriate like `availableTypesSection`?

Reply:
oops, I forgot to rename it